quic/s2n-quic-core/src/crypto/tls/testing.rs

```diff
@@ -170,13 +169,10 @@ impl<S: tls::Session, C: tls::Session> Pair<S, C> {
                     "server should have handshake keys after reading the ClientHello"
                 );
 
-                // TODO remove this when s2n-tls fixes its key schedule
-                if !core::any::type_name::<S>().contains("s2n_quic_tls") {
-                    assert!(
-                        self.server.1.application.crypto.is_some(),
-                        "server should have application keys after reading the ClientHello"
-                    );
-                }
+                assert!(
+                    self.server.1.application.crypto.is_some(),
+                    "server should have application keys after reading the ClientHello"
+                );
```

Comment:
I know this is probably a nit, but this statement made me do a double take every time you sent it to me as an error :) Maybe "after the first exchange"? "after the first round trip"?

Same for the other "after reading the ClientHello"s. Only the early secret is available immediately after the ClientHello. The others are available after the server sends its first flight. The handshake secrets are available after the ServerHello, and the application secrets are available after the ServerFinished.

Reply:
Maybe change these to "server should have handshake keys after sending the ServerHello" and "server should have application keys after sending the ServerFinished".

And for the 3 iteration, you should probably change "client should have application keys after reading the ServerHello" to "client should have application keys after reading the ServerFinished"

Reply:
done in https://github.com/aws/s2n-quic/pull/1228/commits/c18b48a2a25d26030af8e6aef2630e4879690545